### PR TITLE
Add support for the new partitioning customizations to RHEL and CentOS

### PR DESCRIPTION
**Pull request by achilleas-k**
Enabling the new partitioning customizations for RHEL and CentOS.

I also found some duplication in Fedora that I cleaned up.

### CODE REVIEW

**Review by mvo5 — APPROVED** [review]
Thank you for adding the new customizations to all image types.

The duplication makes me slightly sad but I guess there is no way with the current design to do it differently, I would still love to explore ideas here. I was thinking about at least suggesting a comment like "// keep in sync with the other distros" but I guess that is implied already so redundant.

**Comment by achilleas-k** [comment]
> The duplication makes me slightly sad but I guess there is no way with the current design to do it differently

If you're referring to the `checkOptions()` parts, I agree, there's a tonne of duplication there and it's also quite hard to trace.  There have been several attempts to make it nicer, but none of them reached completion.
(Curious what you think about one of my more recent attempts from earlier this year: https://github.com/osbuild/images/compare/main...achilleas-k:images:config-validation).

The way it's structured currently, it's a bit of a necessary evil because for some (but perhaps not most) customizations, the requirements are different between distributions and major versions.  Perhaps extracting the common ones would be enough to get the ball rolling on some of the simplification and to avoid diverging where possible.

**Comment by achilleas-k** [comment]
RHEL 8 aarch64 builds are failing.  :thinking: 

Digging.

File: pkg/distro/rhel/rhel9/options.go [hunk 1]
```diff
@@ -124,19 +124,33 @@ func checkOptions(t *rhel.ImageType, bp *blueprint.Blueprint, options distro.Ima
...
+
+	if err := blueprint.CheckMountpointsPolicy(mountpoints, policies.MountpointPolicies); err != nil {
+		return warnings, err
+	}
+
+	if err := blueprint.CheckDiskMountpointsPolicy(partitioning, policies.MountpointPolicies); err != nil {
+		return warnings, err
+	}
+
+	if err := partitioning.ValidateLayoutConstraints(); err != nil {
```
[the remark this answers is not in the document]
Comment (achilleas-k) [reply pkg/distro/rhel/rhel9/options.go:153]: xref #1065

**Comment by achilleas-k** [comment]
A comment in @ondrejbudai #1080 PR got me thinking:
> This test is only performed on image types with a partition table, thus it doesn't run on e.g. installers and ostree commits.

Instead of listing all supported customizations for each image type like I tried to do in that linked branch, it might be simpler (and more readable) to have properties and capabilities for each image type, and then customizations would be valid or invalid for a given property.  For example "has partition table" is a boolean image type property.  Defining disk or filesystem customizations for any image type that doesn't have a PT is invalid.

This could probably be cleanly extended to all blueprint options: "has package payload" (if false, don't allow packages), "has ostree payload" (requires ostree commit), "can oscap"...

Those payload ones in particular will simplify things a lot.  Right now, when we're dealing with ostree types, we first check for `imageType.RPMOstree` (bool), which is true for all ostree image types.  But that's bad and wrong, because `edge-commit` has a package payload (and therefore installing packages is supported) but `edge-ami` and edge-installer` have an ostree commit payload, so they require an ostree commit and disallow packages.

**Comment by achilleas-k** [comment]
> RHEL 8 aarch64 builds are failing. 🤔
> 
> Digging.

I managed to reproduce the issue.  When building a RHEL 8 image on a Fedora host, it seems the swap partition doesn't get created properly, or at least there's some incompatibility with the RHEL 8 kernel, resulting in:
```
Unable to find swap-space signature
```
when trying to enable the swap space.

Building the same image on RHEL 8 works fine.

It's strange though that this only happens on aarch64.

**Comment by achilleas-k** [comment]
Might be time to switch to running each build on the same host as the target.  Not a huge change to set up, but might need some changes in the test scripts that were written to run on newer python versions.

OTOH, maybe running stuff on RHEL 8 doesn't make sense.  We're not targeting RHEL 8 any more (no more releases), so it might make more sense to test RHEL 8 builds on RHEL 9.

**Comment by mvo5** [comment]
> > RHEL 8 aarch64 builds are failing. 🤔
> > Digging.
> 
> I managed to reproduce the issue. When building a RHEL 8 image on a Fedora host, it seems the swap partition doesn't get created properly, or at least there's some incompatibility with the RHEL 8 kernel, resulting in:
> 
> ```
> Unable to find swap-space signature
> ```
> 
> when trying to enable the swap space.
> 
> Building the same image on RHEL 8 works fine.
> 
> It's strange though that this only happens on aarch64.

I suspect the issue is the pagesize on aarch64. Apparently this got changed recently(ish).c.f. https://bugzilla.redhat.com/show_bug.cgi?id=1978730 and https://bugzilla.redhat.com/show_bug.cgi?id=2001569 - I guess we could try to be smart about it in images too and add a "mkswap -p 64k" option if we target arm and rhel < 9.2(?)

File: pkg/distro/rhel/imagetype.go [hunk 2]
```diff
@@ -193,8 +199,29 @@ func (t *ImageType) GetPartitionTable(
...
+		if options.Size > 0 {
+			// user specified a size on the command line, so let's override the
+			// customization with the calculated/rounded imageSize
+			partitioning.MinSize = imageSize
+		}
+
+		partOptions := &disk.CustomPartitionTableOptions{
+			PartitionTableType: basePartitionTable.Type, // PT type is not customizable, it is determined by the base PT for an image type or architecture
+			BootMode:           t.BootMode(),
+			DefaultFSType:      disk.FS_EXT4, // default fs type for Fedora
```
Comment (mvo5) [inline pkg/distro/rhel/imagetype.go:218]: Hm, this is RHEL here, right? so should this be updated to XFS and comment twekaed?

Comment (achilleas-k) [reply pkg/distro/rhel/imagetype.go:218]: Copy pastaaaaaa!!

**Review by mvo5 — APPROVED** [review]
Thank you

**Pull request merged**